#!/usr/bin/env python
PACKAGE = "pendulum_3d"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("trust",    double_t,    0, "Trust in the user", .0, 0, 1)
gen.add("max_torque", double_t, 0, "Maximum torque applicable by user",    20, 0, 100)
# gen.add("num_links", int_t, 0, "Number of links in the pendulum", 2, 1, 2)

state_enum = gen.enum([ gen.const("Interactive",      int_t, 0, "Only user input to balance pendulum"),
                       gen.const("Controlled_Interactive",     int_t, 1, "User input + feedback controller input")],
                     "An enum to set states")


gen.add("state",    int_t,    0, "A parameter representing the system state (0 - uncontrolled, 1 - controlled)",  1, 0, 1, edit_method=state_enum)



exit(gen.generate(PACKAGE, "pendulum_3d_simulator", "Pendulum"))
